spec/compiler/type_inference/class_var_spec.cr

```diff
@@ -342,4 +342,79 @@ describe "Type inference: class var" do
       ),
       "can't use Int as the type of a class variable yet, use a more specific type"
   end
+
+  it "can find class var in subclass" do
+    assert_type(%(
+      class Foo
+        @@var = 1
+      end
+
+      class Bar < Foo
+        def self.var
+          @@var
+        end
+      end
+
+      Bar.var
+      )) { int32 }
+  end
+
+  it "can find class var through included module" do
+    assert_type(%(
+      module Moo
+        @@var = 1
+      end
+
+      class Bar
+        include Moo
+
+        def self.var
+          @@var
+        end
+      end
+
+      Bar.var
+      )) { int32 }
+  end
+
+  it "errors if redefining class var type in subclass" do
+    assert_error %(
+      class Foo
+        @@x : Int32
+      end
+
+      class Bar < Foo
+        @@x : Float64
+      end
+      ),
+      "class variable '@@x' of Bar is already defined as Int32 in Foo"
+  end
+
+  it "errors if redefining class var type in subclass, with guess" do
```

Comment:
@asterite notice that my comment is about the snippet that does not have type declaration. Only the assignments. github shows the spec before line 393. Maybe we should track in different the type declaration and the assignments without type annotations.